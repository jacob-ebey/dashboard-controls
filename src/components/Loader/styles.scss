@import '../colors.scss';

.loader {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: $color-gray;
  font-family: Consolas, Menlo, Monaco, monospace;
  font-weight: bold;
  font-size: 8em;
  min-width: 360px;
  width: 100%;
  text-align: center;
  opacity: 0.8;

  p {
    color: $color-error;
    font-size: 12px;
  }

  span {
    display: inline-block;
    animation: pulse 0.4s alternate infinite ease-in-out;
    
    &:nth-child(odd) {
      animation-delay: 0.4s;
    }
  }
}

@keyframes pulse {
  to {
    transform: scale(0.8);
    opacity: 0.5;
  }
}