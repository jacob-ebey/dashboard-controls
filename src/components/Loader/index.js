import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.scss';

export default class Component extends React.Component {
  static propTypes = {
    error: PropTypes.string,
  };

  static defaultProps = {
    error: undefined,
  }

  render() {
    const { className, error, children, ...rest } = this.props;
    return (
      <div
        className={`${styles.loader} ${className || ''}`}
        {...rest}
      >
        <span>&#123;</span><span>&#125;</span>
        {error && <br />}
        {error && <p>{error}</p>}
      </div>
    );
  }
}
