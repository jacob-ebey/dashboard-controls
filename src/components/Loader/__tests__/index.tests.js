import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Loader from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('components/Button - index.js', () => {
  const createProps = () => ({
    message: 'ROFLing',
  });

  test('Renders without error', () => {
    const props = createProps();

    const wrapper = shallow(<Loader {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Renders with error', () => {
    const props = {
      ...createProps(),
      error: 'ROFLing error',
    };

    const wrapper = shallow(<Loader {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});