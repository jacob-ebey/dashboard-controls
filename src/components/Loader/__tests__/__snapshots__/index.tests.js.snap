// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button - index.js Renders with error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    error="ROFLing error"
    message="ROFLing"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          {
        </span>,
        <span>
          }
        </span>,
        <br />,
        <p>
          ROFLing error
        </p>,
      ],
      "className": "loader ",
      "message": "ROFLing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "{",
        },
        "ref": null,
        "rendered": "{",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "}",
        },
        "ref": null,
        "rendered": "}",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "ROFLing error",
        },
        "ref": null,
        "rendered": "ROFLing error",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            {
          </span>,
          <span>
            }
          </span>,
          <br />,
          <p>
            ROFLing error
          </p>,
        ],
        "className": "loader ",
        "message": "ROFLing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "{",
          },
          "ref": null,
          "rendered": "{",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "}",
          },
          "ref": null,
          "rendered": "}",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "ROFLing error",
          },
          "ref": null,
          "rendered": "ROFLing error",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`components/Button - index.js Renders without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    error={undefined}
    message="ROFLing"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          {
        </span>,
        <span>
          }
        </span>,
        undefined,
        undefined,
      ],
      "className": "loader ",
      "message": "ROFLing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "{",
        },
        "ref": null,
        "rendered": "{",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "}",
        },
        "ref": null,
        "rendered": "}",
        "type": "span",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            {
          </span>,
          <span>
            }
          </span>,
          undefined,
          undefined,
        ],
        "className": "loader ",
        "message": "ROFLing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "{",
          },
          "ref": null,
          "rendered": "{",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "}",
          },
          "ref": null,
          "rendered": "}",
          "type": "span",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
